flexmeasures/data/tests/test_scheduling_repeated_jobs.py

```diff
@@ -0,0 +1,215 @@
+
+    kwargs = dict(
+        sensor=charging_station,
+        start=start,
+        end=end,
+        belief_time=start,
+        resolution=resolution,
+        flex_model={"soc-at-start": soc_at_start, "soc-targets": soc_targets},
+    )
+    args = []
+
+    hash = hash_function_arguments(args, kwargs)
+
+    # checks that hashes are consistent between calls
+    assert (
+        hash
+        == "LjjneyLDFKRJ6R+v7ZKkOCasaqQDrmqKy2z1gjn7r10=Q3FfKNrQYBgY/qWrw+lEjVEI/GWm9sThyBbOd+sGIKk="
+    )
+
+    # checks that different arguments yield different hashes
+    kwargs2 = copy.deepcopy(kwargs)
+    kwargs2["resolution"] = timedelta(minutes=12)
+
+    hash2 = hash_function_arguments(args, kwargs2)
+
+    assert hash != hash2
+
+
+def test_scheduling_multiple_triggers(
+    caplog, db, app, add_charging_station_assets, setup_test_data
+):
+    caplog.set_level(
+        logging.INFO
+    )  # setting the logging level of the log capture fixture
+
+    soc_at_start = 1
+    target_soc = 5
+    duration_until_target = timedelta(hours=2)
+
+    charging_station = Sensor.query.filter(
+        Sensor.name == "Test charging station"
+    ).one_or_none()
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 2))
+    end = tz.localize(datetime(2015, 1, 3))
+    target_datetime = start + duration_until_target
+    resolution = timedelta(minutes=15)
+    soc_targets = [dict(datetime=target_datetime.isoformat(), value=target_soc)]
+
+    assert (
+        DataSource.query.filter_by(name="FlexMeasures", type="scheduling script")
+        .where()
+        .one_or_none()
+        is None
+    )  # Make sure the scheduler data source isn't there
+
+    # schedule 1 job
+    job1 = create_scheduling_job(
+        sensor=charging_station,
+        start=start,
+        end=end,
+        belief_time=start,
+        resolution=resolution,
+        flex_model={"soc-at-start": soc_at_start, "soc-targets": soc_targets},
+    )
+    work_on_rq(app.queues["scheduling"], exc_handler=exception_reporter)
+
+    caplog.clear()
+
+    # Schedule same job
+    job2 = create_scheduling_job(
+        sensor=charging_station,
+        start=start,
+        end=end,
+        belief_time=start,
+        resolution=resolution,
+        flex_model={"soc-at-start": soc_at_start, "soc-targets": soc_targets},
+    )
+    work_on_rq(app.queues["scheduling"], exc_handler=exception_reporter)
+
+    # checking that the decorator is detecting that the job is repeated
+    assert (
+        sum(
+            [
+                "The function create_scheduling_job has been called alread with the same arguments. Skipping..."
+                in rec.message
+                for rec in caplog.records
+            ]
+        )
+        == 1
+    )
+
+    # checking that they have the same job id
+    assert job1.id == job2.id
+
+    # checking that a different schedule trigger is actually computed
+    soc_at_start = 2
```

Comment:
I like that you are testing against a nested field here. Worth an inline comment.

Reply:
Indeed!